src/GafferArnold/ArnoldLight.cpp

```diff
@@ -65,94 +66,66 @@ ArnoldLight::ArnoldLight( const std::string &name )
 	:	GafferScene::Light( name )
 {
 	storeIndexOfNextChild( g_firstPlugIndex );
-	addChild( new StringPlug( "__shaderName", Plug::In, "", Plug::Default & ~Plug::Serialisable ) );
+
+	addChild( new ArnoldShader( "__shader" ) );
+	addChild( new ShaderPlug( "__shaderIn", Plug::In, Plug::Default & ~Plug::Serialisable ) );
+
+	shaderNode()->addChild( new Plug( "out", Plug::Out ) );
```

Comment:
Do we need this, or can we allow `ArnoldShader::loadShader()` to make it for us later? I see we're repeating the `shaderInPlug()->setInput( shaderNode()->outPlug() );` call in `loadShader()`, which implies maybe yes?

Then again, maybe we need it for compatibility with pre-0.45 scripts, where we didn't serialise a `loadShader()` call and instead serialised a bunch of `light["parameters"].addChild()` calls. Do we have test coverage for this?

Reply:
iirc, something was stomping on the `out` plug when the shader was loaded (deleted and re-made) so the internal connection was broken, which is where the re-connection came from. I can look further to fixing the source if that's desired, but that seemed like it might have wider ramifications?